app/abilities/person_readables.rb

```diff
@@ -50,6 +50,15 @@ def group_accessible_people
   def accessible_people
     if user.root?
       Person.only_public_data
+    elsif Ability.new(user).can?(:show_people_without_role, Person)
+      join_fragment = "INNER JOIN roles AS roles0 ON roles0.person_id = people.id " \
+                      "INNER JOIN groups AS groups0 ON groups0.id = roles0.group_id"
+      ac = accessible_conditions.to_a
+      ac[0] = ac[0].gsub('groups', 'groups0').gsub('roles', 'roles0')
+      Person.only_public_data.
+             joins(join_fragment).
+             where(ac).
+             uniq
```

Comment:
[Hier](https://github.com/hitobito/hitobito/blob/master/app/domain/person/list_filter.rb#L36) werden die `PersonReadables` ebenfalls ohne group aufgerufen. In diesem Fall sollen die gelöschten jedoch nicht enthalten sein. Es braucht wohl also doch ein zusätzliches Query für die gelöschten im `FullTextController`. Immerhin müssen wir uns dafür nicht mehr den Kopf über diese Join Bastelei zerbrechen.